docs/aws/audit/appsyncmonitoring/rules/appsync_logging_enabled_remediation.mdx

```diff
@@ -2,6 +2,153 @@
+<Accordion title='Using CLI'>
+To prevent the misconfiguration of "AppSync API Should Have Logging Enabled" in AWS AppSync using AWS CLI, follow these steps:
+
+1. **Create a CloudWatch Log Group:**
+   Ensure you have a CloudWatch Log Group where the AppSync logs will be stored. If you don't have one, create it using the following command:
+   ```sh
+   aws logs create-log-group --log-group-name <your-log-group-name>
+   ```
+
+2. **Create an IAM Role for AppSync Logging:**
+   Create an IAM role that AppSync can assume to write logs to CloudWatch. Use the following command to create the role:
+   ```sh
+   aws iam create-role --role-name AppSyncLoggingRole --assume-role-policy-document file://trust-policy.json
+   ```
+   The `trust-policy.json` should contain:
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "Service": "appsync.amazonaws.com"
+         },
+         "Action": "sts:AssumeRole"
+       }
+     ]
+   }
+   ```
+
+3. **Attach Policy to the IAM Role:**
+   Attach a policy to the IAM role that allows it to write logs to CloudWatch. Use the following command:
+   ```sh
+   aws iam put-role-policy --role-name AppSyncLoggingRole --policy-name AppSyncLoggingPolicy --policy-document file://logging-policy.json
+   ```
+   The `logging-policy.json` should contain:
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Action": [
+           "logs:CreateLogStream",
+           "logs:PutLogEvents"
+         ],
+         "Resource": "arn:aws:logs:<region>:<account-id>:log-group:<your-log-group-name>:*"
+       }
+     ]
+   }
+   ```
+
+4. **Enable Logging for AppSync API:**
+   Finally, enable logging for your AppSync API by updating the API settings to include the CloudWatch Log Group and the IAM Role. Use the following command:
+   ```sh
+   aws appsync update-graphql-api --api-id <your-api-id> --log-config cloudWatchLogsRoleArn=arn:aws:iam::<account-id>:role/AppSyncLoggingRole,fieldLogLevel=ALL,excludeVerboseContent=false
+   ```
+
+By following these steps, you ensure that logging is enabled for your AppSync API, which helps in monitoring and troubleshooting.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of AppSync API not having logging enabled in AWS using Python scripts, you can follow these steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed, which is the AWS SDK for Python. You can install it using pip if you haven't already.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Set up a Boto3 session to interact with AWS services. You need to configure your AWS credentials and region.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>